aot.py

```diff
@@ -23,9 +23,14 @@ def broadcast(x):
     return x
 
 def all_reduce(x):
-    dist.all_reduce(x)
+    # AOT in general don't support in-place functions since
+    # it actually executes the function twice: one for shape inference,
+    # and one for tracing during the first time.
+    # So, let's work around by clone the input.
+    xc = x.clone()
```

Comment:
Or you might endorse a more general fix in https://github.com/pytorch/functorch/pull/802